### DISCUSSION TITLE
SFDX: Create Apex Class command should allow you to specify a template.

### DISCUSSION THREAD

**Discussion by joseph-ortiz** [forcedotcom/salesforcedx-vscode#5375] [Ideas]
**Is your feature request related to a problem? Please describe.**
If you use the command palette to create an apex class, the DefaultApexClass template is used without an option to select what template.

**Describe the solution you'd like**
I am proposing an additional input in the "SFDX:Create Apex Class" command to select a template. By default it should be DefaultApexClass but I'd like to specify the template sometimes. If there is a setting to toggle this feature on/off from vscode settings is a possibility for those who do not want to select a template.

1. Open VSCode Command Pallete
2. select the command, "SFDX:Create Apex Class"
3. Enter desired apex class name. Hit enter.
4 if applicable, Enter desired directory. Hit enter.
5. Type/Select ApexClass template you want to use. Hit enter.
6. Class is created.

**Describe alternatives you've considered**
i can just use the SFDX cli command with the template flag to create the class .

**Additional context**

**Comment by mshanemc** [comment]
done (but we do template first)
